ably/realtime_channel.go

```diff
@@ -896,6 +918,86 @@ func (c *RealtimeChannel) notify(msg *protocolMessage) {
 	}
 }
 
+// processMessageWithDelta processes a message with delta decoding support (RTL19, RTL20).
+func (c *RealtimeChannel) processMessageWithDelta(msg *Message) error {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+
+	// Update decoding context with current state (RTL19, RTL20)
+	c.decodingContext.BasePayload = c.basePayload
+	c.decodingContext.LastMessageID = c.lastMessageID
+
+	// Decode message with delta support
+	var cipher channelCipher
+	if c.options != nil {
+		cipher, _ = (*protoChannelOptions)(c.options).GetCipher()
+	}
+	decodedMsg, newBasePayload, err := msg.withDecodedDataAndContext(cipher, c.decodingContext)
+	if err != nil {
+		// Check for specific delta-related errors
+		if containsErrorCode(err, 40018) || containsErrorCode(err, 40019) {
+			return err
+		}
+		// For other errors, just log and continue (RSL6b)
+		c.client.log().Warnf("Message decode error on channel %q: %v", c.Name, err)
+		// Still update the message with partial decoding
+		*msg = decodedMsg
+		return nil
+	}
+
+	// Update message with decoded data
+	*msg = decodedMsg
```

Comment:
It looks like `withDecodedDataAndContext` returns the same message, but just mutates that message's `Data` field, so this here seems redundant, `msg` and `decodedMsg` point to the same thing.